azavea/temperate
# Cleanup Area Indicators project - Make queries work

## Overview

Originally this task was to fix #1235 but it turns out that wasn't an issue with the newly ingested layer, which is fantastic news. Instead, I took the time to clean up a few things:

- Move Operations and associated data retrieval classes
  back into a sub of the api package. These operations
  are much more closely related to the API query operations
  we'll make rather than the datamodel.
- Fix bug where we were returning a MapCellService instead
  of a DataService
- Move all *Service classes into api subpackage
- Add Config class for retrieving and accessing values set
  by env variables, and make changes throughout code to use
  these values
- Update default gt layer bucket + path and test output
- Delete old commented out Router sample code
- Move typealiases defined on Operations object to datamodel
  package object so that we can move Operations object back
  to api package
- Print queried area as GeoJson to the log for easier debugging

### Demo

<img width="719" alt="Screen Shot 2019-08-28 at 11 32 03 AM" src="https://user-images.githubusercontent.com/1818302/63878671-153c5c00-c998-11e9-95ba-7b8394f7bb18.png">

### Notes

The query always just pulls and aggregates band zero of a MultibandTile where each tile is the data for a given ClimateModel + Variable for a given time slice (day). A separate PR will handle properly querying for all the requested data given a set of indicator parameters. This was enough of a refactor, bugfix, cleanup that I wanted to get it in place for other parallel efforts.

## Testing Instructions

Ensure your terminal is setup with the correct AWS_PROFILE set, then:
`./sbt "api/run"`

And make a query such as:
http://localhost:8108/climate-data/rcp85/indicator/average_high_temperature?years=2020:2021

The demo layer only has data for 2020 so you can't query any other range. You can query for other Indicators but you'll always get approximately the same value due to #1272 


Closes #1235
